### PR TITLE
Cuda implementation of SetPixel() function

### PR DESCRIPTION
**Pull request by gkarlos**
As the title suggests, this PR solves #538 .  Unit tests included.

### CODE REVIEW

**Comment by gkarlos** [comment]
Looks like unit_test times out on the osx build? Any ideas why that might be?

**Comment by ihhub** [comment]
@gkarlos I've restarted job. It might be time limit for pipeline job as 2 projects on my account receive many commits which trigger compilation on virtual machine. I'm working on reducing time for virtual machines.

File: .gitignore [hunk 1]
```diff
@@ -55,10 +55,3 @@
 *.fatbin
 *.cu.deps
 *.cu.cache
-
-
-#build 
-/build/*
-
-#editors/IDEs
-/.vscode/*
```
Comment (gkarlos) [inline .gitignore]: I know these are unrelated to issue #538, but i think it'd be good if .gitignore included them. It's good not having to worry about staging build for instance. Are you interested in another PR that adds some common stuff to the .gitignore?

Comment (ihhub) [reply .gitignore]: @gkarlos please create a separate PR for this change if you think that this is valid :)

**Comment by ihhub** [comment]
@gkarlos thank you very much for the contribution to the project!

**Pull request merged**